ihmc-pub-sub/src/main/java/us/ihmc/pubsub/attributes/ParticipantProfile.java

```diff
@@ -136,112 +136,136 @@ public ParticipantProfile discoveryServer(String discoveryServerAddress, int dis
    }
 
    /**
-    * Bind this participant to only the addresses in bindToAddressRestrictions
-    * 
-    * Functionality, this will create a new UDPv4 transport with the whitelist set to "bindToAddressRestrictions". Optionally, a shared memory transport will be added as well.
-    * useBuiltinTransports will be set to false 
-    * 
-    * @param addSharedMemoryTransport Enabled shared memory communication by adding a shared memory transport to this participant.
-    * @param bindToAddressRestrictions Limit the scope of this participant to the list of hosts. If null or empty, the participant will not be able to communicate.
-    * @return
+    * Add transport to use with this participant and register it to this participant
     */
-   public ParticipantProfile bindToAddressRestrictions(boolean addSharedMemoryTransport, List<InetAddress> bindToAddressRestrictions)
+   public ParticipantProfile addTransport(TransportDescriptorType transport)
```

Comment:
See this example for understanding the difference between transport_descriptors and userTransports
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<dds xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <profiles>
        <transport_descriptors>
            <transport_descriptor>
                <transport_id>dc261415-468c-4302-9da6-f1e3ead3c9a3</transport_id>
                <type>SHM</type>
            </transport_descriptor>
            <transport_descriptor>
                <transport_id>41087b0d-7915-4c1d-a276-57de9a749fea</transport_id>
                <type>UDPv4</type>
            </transport_descriptor>
        </transport_descriptors>
        <participant profile_name="66a19794-8016-4736-b280-32930be03091">
            <domainId>219</domainId>
            <rtps>
                <name>srt_stream_image_message_republisher</name>
                <builtin>
                    <discovery_config>
                        <leaseDuration>
                            <nanosec>-1</nanosec>
                            <sec>2147483647</sec>
                        </leaseDuration>
                    </discovery_config>
                </builtin>
                <userTransports>
                    <transport_id>dc261415-468c-4302-9da6-f1e3ead3c9a3</transport_id>
                    <transport_id>41087b0d-7915-4c1d-a276-57de9a749fea</transport_id>
                </userTransports>
                <useBuiltinTransports>false</useBuiltinTransports>
            </rtps>
        </participant>
    </profiles>
</dds>
```